File: snippets/dart_snippets/lib/list_payments.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<List<Payment>> listPayments() async {
  // ANCHOR: list-payments
  ListPaymentsRequest req = ListPaymentsRequest();
  List<Payment> paymentsList = await breezSDK.listPayments(req: req);
  print(paymentsList);
  // ANCHOR_END: list-payments
  return paymentsList;
}

Future<List<Payment>> listPaymentsFiltered({
  int? fromTimestamp,
  int? toTimestamp,
  bool? includeFailures,
  int? offset,
  int? limit,
}) async {
  // ANCHOR: list-payments-filtered
  /// Get the desired epoch timestamp in seconds
  int fromTimestamp = DateTime.now().subtract(const Duration(minutes: 30)).millisecondsSinceEpoch ~/ 1000;

  ListPaymentsRequest req = ListPaymentsRequest(
    filters: [PaymentTypeFilter.Sent],
    fromTimestamp: fromTimestamp,
    toTimestamp: toTimestamp,
    includeFailures: includeFailures,
    offset: offset,
    limit: limit,
  );
  List<Payment> paymentsList = await breezSDK.listPayments(req: req);
  print(paymentsList);
  // ANCHOR_END: list-payments-filtered
  return paymentsList;
}

File: snippets/swift/BreezSDKExamples/Sources/ListPayments.swift
//
//  ListPayments.swift
//
//
//  Created by ruben on 13/11/2023.
//

import BreezSDK
import Foundation

func ListPayments(sdk: BlockingBreezServices) -> [Payment]? {
    // ANCHOR: list-payments
    let payments = try? sdk.listPayments(req: ListPaymentsRequest())
    // ANCHOR_END: list-payments
    return payments
}

func ListPaymentsFiltered(sdk: BlockingBreezServices) -> [Payment]? {
    // ANCHOR: list-payments-filtered
    let payments = try? sdk.listPayments(
        req: ListPaymentsRequest(
            filters: [.sent],
            fromTimestamp: 1_696_880_000,
            includeFailures: true
        ))
    // ANCHOR_END: list-payments-filtered
    return payments
}
